src/cdk/scripts/retrieve-config.sh

```diff
@@ -0,0 +1,292 @@
+        user_arn=$(echo "$identity_output" | jq -r '.Arn' 2>/dev/null || echo "unknown")
+        log_debug "AWS Account ID: $account_id"
+        log_debug "AWS User/Role ARN: $user_arn"
+    else
+        log_error "AWS credentials validation failed:"
+        log_error "$identity_output"
+        log_error ""
+        log_error "Please configure AWS credentials using one of:"
+        log_error "  - AWS CLI: aws configure"
+        log_error "  - Environment variables: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY"
+        log_error "  - IAM roles (for EC2/ECS/Lambda)"
+        log_error "  - AWS SSO: aws sso login"
+        exit 1
+    fi
+
+    # Check current region
+    local current_region
+    current_region=$(aws configure get region 2>/dev/null || echo "${AWS_REGION:-not set}")
+    log_debug "AWS Region: $current_region"
+}
+
+# Parameter retrieval with detailed error handling
+retrieve_parameter() {
+    log_info "=============================================="
+    log_info "Retrieving configuration from Parameter Store"
+    log_info "Parameter name: $PARAMETER_NAME"
+    log_info "Target file: $TARGET_ENV_FILE"
+    log_info "=============================================="
+
+    log_debug "Attempting to retrieve parameter from SSM..."
+
+    local ssm_output
+    local ssm_error
+    local exit_code
+
+    # Capture both stdout and stderr
+    set +e  # Temporarily disable exit on error
+    {
+        ssm_output=$(aws ssm get-parameter \
+            --name "$PARAMETER_NAME" \
+            --with-decryption \
+            --query "Parameter.Value" \
+            --output text 2>&1)
+        exit_code=$?
+    }
+    set -e  # Re-enable exit on error
+
+    log_debug "AWS SSM command exit code: $exit_code"
+
+    if [ $exit_code -eq 0 ] && [ -n "$ssm_output" ] && [ "$ssm_output" != "None" ]; then
+        # Success case
+        log_info "Parameter retrieved successfully"
+        log_debug "Parameter content length: ${#ssm_output} characters"
+
+        # Validate content format (basic checks)
+        if [[ "$ssm_output" == *"="* ]]; then
+            log_debug "Content appears to contain environment variable assignments"
+        else
+            log_warn "Content does not appear to contain typical .env format (no '=' found)"
+        fi
+
+        # Write to file
+        echo "$ssm_output" > "$TARGET_ENV_FILE"
+
+        if [ -f "$TARGET_ENV_FILE" ]; then
+            local file_size
+            file_size=$(wc -c < "$TARGET_ENV_FILE" 2>/dev/null || echo "0")
+            log_info "Configuration file created: $TARGET_ENV_FILE ($file_size bytes)"
+
+            # Show content (be careful with sensitive data)
+            if [[ "$LOG_LEVEL" == "DEBUG" ]]; then
+                log_debug "Configuration file contents:"
+                log_debug "----------------------------------------------"
+                cat "$TARGET_ENV_FILE" | sed 's/\(.*=\).*/\1[REDACTED]/' >&2
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Unnecessary use of &#39;cat&#39; in shell scripts, can lead to inefficient code and reduced script performance. Instead of &#39;cat file | command&#39;, use input redirection &#39;command &lt; file&#39; or pass the filename directly `command file` when possible. This practice improves script efficiency, especially for commands that benefit from seekable input. It also enhances code readability and maintainability. For more information on secure coding practices, including shell scripting, refer to the OWASP Secure Coding Practices Quick Reference Guide: https://owasp.org/www-project-secure-coding-practices-quick-reference-guide/